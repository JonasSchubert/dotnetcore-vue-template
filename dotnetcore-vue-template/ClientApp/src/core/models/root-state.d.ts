import { ExamplesState } from '@/examples/models';
import { BaseState } from "@/libs/core/models";
import { I18nState } from "@/libs/i18n/models";
import { LocalStorageState } from "@/libs/local-storage/models";

export interface RootState extends BaseState {
  examples?: ExamplesState;
  i18n?: I18nState;
  localStorage?: LocalStorageState;
}
