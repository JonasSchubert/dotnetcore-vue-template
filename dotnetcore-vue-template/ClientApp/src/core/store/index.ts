import Vue from 'vue';
import Vuex, { Store, StoreOptions } from 'vuex';
import { createI18nModule } from '@/core/i18n/module-helper';
import { RootState } from '@/core/models';
import { createModule as createExamplesModule } from '@/examples/store';
import { ModuleType as ExamplesModule } from '@/examples/store/types';
import { ModuleType as LocalStorageModule } from "@/libs/local-storage/store/types";
import { createLocalStorageModule } from "@/libs/local-storage/store";
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

Vue.use(Vuex);

export const createStore = (vuetify: any): Store<RootState> => {
  const store: StoreOptions<RootState> = {
    getters: createGetters(),
    modules: {
      [ExamplesModule]: createExamplesModule(),
      i18n: createI18nModule(),
      [LocalStorageModule]: createLocalStorageModule<RootState>('dotnetcore-vue-template', vuetify)
    },
    mutations: createMutations(),
    state: createState(),
    strict: process.env.NODE_ENV !== 'production'
  };

  return new Store<RootState>(store);
};
