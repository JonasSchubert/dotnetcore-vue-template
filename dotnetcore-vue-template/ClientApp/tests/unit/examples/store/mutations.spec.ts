import { Example, ExamplesState } from '@/examples/models';
import { createMutations } from '@/examples/store/mutations';
import { MutationTypes } from '@/examples/store/types';

describe('examples/mutations', () => {
  let state: ExamplesState;
  const mutations = createMutations();

  beforeEach(() => {
    state = {
      error: undefined,
      isLoading: false,
      list: []
    };
  });

  describe(`${MutationTypes.setError}`, () => {
    test('should set error', () => {
      // Arrange
      const error: Error = {
        message: 'message',
        name: 'name'
      };

      // Act
      mutations[MutationTypes.setError](state, { error });

      // Assert
      expect(state.error).toMatchSnapshot({
        message: 'message',
        name: 'name'
      });
    });
  });

  describe(`${MutationTypes.setIsLoading}`, () => {
    test('should set isLoading', () => {
      // Arrange
      const isLoading = true;

      // Act
      mutations[MutationTypes.setIsLoading](state, { isLoading });

      // Assert
      expect(state.isLoading).toBeTruthy();
    });
  });

  describe(`${MutationTypes.setList}`, () => {
    test('should set list', () => {
      // Arrange
      const list: Example[] = [
        { id: '1' } as Example
      ];

      // Act
      mutations[MutationTypes.setList](state, { list });

      // Assert
      expect(state.list).toMatchSnapshot([
        { id: '1' }
      ]);
    });
  });
});
