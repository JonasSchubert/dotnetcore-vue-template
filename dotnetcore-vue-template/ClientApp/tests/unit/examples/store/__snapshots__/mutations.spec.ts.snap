// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/mutations setError should set error 1`] = `
Object {
  "message": "message",
  "name": "name",
}
`;

exports[`examples/mutations setList should set list 1`] = `
Object {
  "0": Object {
    "id": "1",
  },
}
`;
